### PR TITLE
[IMP] fleet: Add analytic account and mobility card

### PR DESCRIPTION
**Pull request by jbm-odoo**
Create analytic account from fleet (and link it to vehicle) based
on the plate and the fiscal deduction rate (if l10n_hr_payroll_fleet
is installed).

Add on employee a Mobility Card number. This number is set on vehicle
when we put a partner linked to employee on a vehicle.

taskID: 2127641

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/fleet/models/fleet_vehicle.py [hunk 1]
```diff
@@ -156,6 +156,10 @@ def _compute_contract_reminder(self):
             record.contract_renewal_total = total - 1  # we remove 1 from the real total for display purposes
             record.contract_renewal_name = name
 
+    def _get_analytic_name(self):
+        # This function is used in fleet_account and is overrided in l10n_be_hr_payroll_fleet
+        return self.license_plate or 'No plate'
```
Comment (tivisse) [inline addons/fleet/models/fleet_vehicle.py]:
```suggestion
        return self.license_plate or _('No plate')
```

File: addons/fleet_account/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+{
+    'name': 'Fleet - Account',
+    'version': '0.1',
```
Comment (tivisse) [inline addons/fleet_account/__manifest__.py]:
```suggestion
    'version': '1.0',
```
Nope ? :)

File: addons/fleet_account/models/fleet_vehicle.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+from odoo import fields, models, _
+from odoo.exceptions import AccessError
+
+
+class FleetVehicle(models.Model):
+    _inherit = 'fleet.vehicle'
+
+    analytic_account_id = fields.Many2one('account.analytic.account', string='Analytic Account', copy=False, ondelete='set null',
+        domain="['|', ('company_id', '=', False), ('company_id', '=', company_id)]", check_company=True,
```
Comment (tivisse) [inline addons/fleet_account/models/fleet_vehicle.py]: I think that check_company actually adds your domain. See: https://github.com/odoo/odoo/blame/master/odoo/fields.py#L2332

File: addons/fleet_account/models/fleet_vehicle.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+        help="Analytic account to which this vehicle is linked for financial management.")
+
+    def action_analytic_account(self):
```
Comment (tivisse) [inline addons/fleet_account/models/fleet_vehicle.py]:
```suggestion
    def action_generate_analytic_account(self):
```

File: addons/fleet_account/models/fleet_vehicle.py [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+        if not self.env.user.has_group('fleet.fleet_group_user'):
+            raise AccessError(_("Do not have access, you must be at lease fleet user."))
```
Comment (tivisse) [inline addons/fleet_account/models/fleet_vehicle.py]:
```suggestion
            raise AccessError(_("Sorry, you must be at least a fleet user to make this action."))
```

File: addons/hr_expense/models/hr_employee.py [hunk 6]
```diff
@@ -30,6 +30,8 @@ def _compute_expense_manager(self):
             manager = employee.parent_id.user_id
             if manager and manager.has_group('hr_expense.group_hr_expense_user') and (employee.expense_manager_id == previous_manager or not employee.expense_manager_id):
                 employee.expense_manager_id = manager
+            elif not employee.expense_manager_id:
```
Comment (tivisse) [inline addons/hr_expense/models/hr_employee.py]: Mmmh. Seems louche nope ?

File: addons/hr_holidays/models/hr_employee.py [hunk 7]
```diff
@@ -158,6 +158,8 @@ def _compute_leave_manager(self):
             manager = employee.parent_id.user_id
             if manager and employee.leave_manager_id == previous_manager or not employee.leave_manager_id:
                 employee.leave_manager_id = manager
+            elif not employee.leave_manager_id:
+                employee.leave_manager_id = False
```
Comment (tivisse) [inline addons/hr_holidays/models/hr_employee.py]: I suppose that it's to set a value in all the cases, but it seems louche.
Why not:
``` python
else:
    employee.leave_manager_id = False
```

File: addons/fleet_account/__manifest__.py (hunk 2 above)
Comment (jbm-odoo) [reply addons/fleet_account/__manifest__.py]: I took the same as Fleet, but 1.0 is better

**Comment by jbm-odoo** [comment]
@tivisse I fixed problems :)

**Comment by tivisse** [comment]
@jbm-odoo Thanks ;)

@robodoo rebase-ff r+

**Pull request closed without merging**